src/Http/Http.Abstractions/src/Routing/RouteValueDictionary.cs

```diff
@@ -753,12 +755,12 @@ public PropertyStorage(object value)
 
             // Cache the properties so we can know if we've already validated them for duplicates.
             var type = Value.GetType();
-            if (!_propertyCache.TryGetValue(type, out Properties!))
+            Properties = _propertyCache.GetValue(type, static x =>
```

Comment:
Can you run some performance tests on this?

Reply:
I ran the following benchmark comparing the retreival speed between the two collections:

``` csharp
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using System.Collections.Concurrent;
using System.Runtime.CompilerServices;

BenchmarkRunner.Run<ConditionalWeakTableBenchmark>(args: args);

[RankColumn]
public class ConditionalWeakTableBenchmark
{
    private readonly ConditionalWeakTable<object, object> _cwt = new();
    private readonly ConcurrentDictionary<object, object> _dict = new();
    private readonly object _key = new();

    [Benchmark]
    public object ConditionalWeakTable() =>
        _cwt.GetValue(_key, x => new object());

    [Benchmark]
    public object ConcurrentDictionary()
    {
        if (!_dict.TryGetValue(_key, out var value))
        {
            var newValue = new object();
            _dict.TryAdd(_key, newValue);
            return newValue;
        }
        return value;
    }
}
```
And here are the results:

|               Method |     Mean |    Error |   StdDev | Rank |
|--------------------- |---------:|---------:|---------:|-----:|
| ConditionalWeakTable | 15.04 ns | 0.289 ns | 0.256 ns |    1 |
| ConcurrentDictionary | 15.43 ns | 0.289 ns | 0.271 ns |    2 |

I ran them more than once and the results were similar. `ConditionalWeakTable` wins by a negligible difference. In case you don't know, its `GetValue` method uses double-checked locking and replaces the entire underlying hash table if it gets full, prividing good steady-state performance.